ldes-server-compaction/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/compaction/application/services/CompactionCandidateService.java

```diff
@@ -0,0 +1,60 @@
+package be.vlaanderen.informatievlaanderen.ldes.server.compaction.application.services;
+
+import be.vlaanderen.informatievlaanderen.ldes.server.compaction.domain.entities.ViewCapacity;
+import be.vlaanderen.informatievlaanderen.ldes.server.domain.model.LdesFragmentIdentifier;
+import be.vlaanderen.informatievlaanderen.ldes.server.fetching.entities.CompactionCandidate;
+import be.vlaanderen.informatievlaanderen.ldes.server.fetching.repository.AllocationRepository;
+import be.vlaanderen.informatievlaanderen.ldes.server.fragmentation.repository.FragmentRepository;
+import one.util.streamex.StreamEx;
+import org.springframework.stereotype.Component;
+
+import java.util.*;
+import java.util.concurrent.atomic.AtomicInteger;
+
+@Component
+public class CompactionCandidateService {
+	private final AllocationRepository allocationRepository;
+	private final FragmentRepository fragmentRepository;
+
+
+	public CompactionCandidateService(AllocationRepository allocationRepository, FragmentRepository fragmentRepository) {
+		this.allocationRepository = allocationRepository;
+		this.fragmentRepository = fragmentRepository;
+	}
+
+	public Map<Integer, Set<CompactionCandidate>> getCompactionTaskList(ViewCapacity viewCapacity) {
```

Comment:
Why do you return a map? The name of the method literally says "list" and that is all you are interested in.